src/jit/gentree.cpp

```diff
@@ -13758,11 +13759,14 @@ void ReturnTypeDesc::Initialize(Compiler* comp, CORINFO_CLASS_HANDLE retClsHnd)
         }
     }
 
+#elif defined(_TARGET_X86_)
+    m_regType0 = TYP_INT;
+    m_regType1 = TYP_INT;
```

Comment:
Here we are making an assumption on x86 that ReturnTypeDesc is always used for TYP_LONG return types.  But eventually we want to use ReturnTypeDesc for describing all return types and not just long.

How easy or difficult to check that the type handle passed corresponds to that of TYP_LONG/TYP_ULONG?
I would suggest minimum to add a Assumption/TODO comment here.